### PR TITLE
Upgrade to Maven Resolver 1.9.7 and Maven Resolver Provider 3.9.1

### LINKED ISSUE #1399: Upgrade to Maven Resolver 1.9.7

### PR DESCRIPTION
**Pull request by prithvitewatia**
This PR closes issues #1399 and #1400.

Maven resolver provider 3.9.1 has dependency maven resolver 1.9.7. Thus, this PR may close both the issues.

### CODE REVIEW

File: initializr-version-resolver/src/main/java/io/spring/initializr/versionresolver/MavenResolverDependencyManagementVersionResolver.java [hunk 1]
```diff
@@ -81,6 +82,15 @@ class MavenResolverDependencyManagementVersionResolver implements DependencyMana
 		LocalRepository localRepository = new LocalRepository(cacheLocation.toFile());
 		this.repositorySystem = serviceLocator.getService(RepositorySystem.class);
 		session.setLocalRepositoryManager(this.repositorySystem.newLocalRepositoryManager(session, localRepository));
+		final Properties systemProperties = new Properties();
```
Comment (snicoll) [inline initializr-version-resolver/src/main/java/io/spring/initializr/versionresolver/MavenResolverDependencyManagementVersionResolver.java:85]: Can you explain this change and why you think it's necessary. I remember that something was off so I'd like to understand how you found out about this.

**Review by snicoll — COMMENTED** [review]
Thanks for the PR! I've left a comment for you when you have time.

**Comment by prithvitewatia** [comment]
Sure,

When we upgrade from maven provider 3.8.7 to 3.9.1, we get ModelBuildingException as you mentioned in #1399.
This exception occurs at line 65 in [JdkVersionProfileActivator class](https://github.com/apache/maven/blob/maven-3.9.1/maven-model-builder/src/main/java/org/apache/maven/model/profile/activation/JdkVersionProfileActivator.java) when the value for 
`java.version` is accessed from system properties. 

In version 3.8.7, system properties value was already being set in [MavenRepositorySystemUtils 3.8.7](https://github.com/apache/maven/blob/maven-3.8.7/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java) but were not being set in [MavenRepositorySystemUtils 3.9.1](https://github.com/apache/maven/blob/maven-3.9.1/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java).

Thus I simply added the missing system properties in the way originally present in MavenRepositorySystemUtils 3.8.7.

File: initializr-version-resolver/src/main/java/io/spring/initializr/versionresolver/MavenResolverDependencyManagementVersionResolver.java (hunk 1 above)
Comment (prithvitewatia) [reply initializr-version-resolver/src/main/java/io/spring/initializr/versionresolver/MavenResolverDependencyManagementVersionResolver.java:85]: Sure,

When we upgrade from maven provider 3.8.7 to 3.9.1, we get ModelBuildingException as you mentioned in https://github.com/spring-io/initializr/issues/1399.
This exception occurs at line 65 in [JdkVersionProfileActivator class](https://github.com/apache/maven/blob/maven-3.9.1/maven-model-builder/src/main/java/org/apache/maven/model/profile/activation/JdkVersionProfileActivator.java) when the value for
java.version is accessed from system properties.

In version 3.8.7, system properties value was already being set in [MavenRepositorySystemUtils 3.8.7](https://github.com/apache/maven/blob/maven-3.8.7/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java) but were not being set in [MavenRepositorySystemUtils 3.9.1](https://github.com/apache/maven/blob/maven-3.9.1/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java).

Thus I simply added the missing system properties in the way originally present in MavenRepositorySystemUtils 3.8.7.

**Comment by snicoll** [comment]
@prithvitewatia thank you for making your first contribution to Spring Initializr.

**Pull request closed without merging**